core/model/admin.py

```diff
@@ -62,11 +62,16 @@ def password(self):
         raise NotImplementedError("Password comparison is only with Admin.authenticate")
 
     @password.setter
-    def password(self, value):
-        self.password_hashed = str(bcrypt.hashpw(value, bcrypt.gensalt()))
-
-    def has_password(self, password):
-        return self.password_hashed == bcrypt.hashpw(password, self.password_hashed)
+    def password(self, value: str) -> None:
+        self.password_hashed = bcrypt.hashpw(value.encode(), bcrypt.gensalt()).decode()
+
+    def has_password(self, password: str) -> bool:
+        if self.password_hashed is None:
+            return False
+        return (
+            self.password_hashed
+            == bcrypt.hashpw(password.encode(), self.password_hashed.encode()).decode()
```

Comment:
Not that important but I think this line could be a bit simplified:
`return bcrypt.checkpw(password.encode(), self.password_hashed.encode())`